plugin/core/windows.py

```diff
@@ -497,8 +466,17 @@ def start(self, config: ClientConfig, initiating_view: sublime.View) -> None:
         self._window.status_message("Starting " + config.name + "...")
         try:
             workspace_folders = sorted_workspace_folders(self._workspace.folders, file_path)
-            session_type = get_session_type(config.name)
-            session = session_type(self, self._settings, workspace_folders, config)
+            plugin_class = get_plugin(config.name)
+            if plugin_class is not None:
+                if plugin_class.needs_update_or_installation():
+                    # TODO: Run on separate thread somewhere
+                    plugin_class.install_or_update()
```

Comment:
I'll just mark that for myself.
Will be great when we won't even attempt to start the session when server is installing.